src/content/reference/react/cache.md

```diff
@@ -0,0 +1,443 @@
+---
+title: cache
+canary: true
+---
+
+<Canary>
+`cache` is currently only available in React’s [Canary](https://react.dev/community/versioning-policy#canary-channel) and [experimental](https://react.dev/community/versioning-policy#experimental-channel) channels. Please ensure you understand the limitations before using `cache` in production.
+
+Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+
+</Canary>
+
+<Intro>
+
+`cache` lets you cache the result of a data fetch or computation.
+
+```js
+const cachedFn = cache(fn);
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `cache(fn)` {/*cache*/}
+
+Wrap a function with `cache` to receive a version of that function with caching.
```

Comment:
Borrowing phrasing from the other documentation pages (e.g. `createContext`), how about:

> Call `cache` outside of any components to create a version of the function with caching.

In particular, I find the "Call [...] outside of any components" to be pretty valuable in setting the expectation about where in a developer's code this function is intended to be used.

---

I also noticed that we refer to the behavior as "caching" instead of "memoization".

I understand if the intent is to make this more accessible to developers, but I couldn't help but notice that [the documentation for `memo`](https://react.dev/reference/react/memo) already uses "memoization".

Reply:
Yea `memo` does, but they do also define memoization https://react.dev/reference/react/memo#skipping-re-rendering-when-props-are-unchanged. This is how `useMemo` also does it: https://react.dev/reference/react/useMemo#caveats

I think the lack of "memoize" in the early section, before we define it, feels sufficient, is there something that doesn't to you?

Reply:
Updated to use similar wording to `createContext`